ethdb/database.go

```diff
@@ -154,11 +160,19 @@ type Writer interface {
 	AncientWriter
 }
 
+// Stater contains the methods required to retrieve states from both key-value as well as
```

Comment:
I don't really understand the word`Stater`. It either makes me think of the `Stat` method on files, or basically "the ability to read state (trie)-data". What connotation is it that you want express?

Reply:
`Stater` means we can retrieve some basic database information from. Like the leveldb statistics. And I extend it to the ancient store so that we can retrieve the "ancient directory" back, it's also a kind of database information.

Reply:
Perhaps we can figure out some better name. Don't have any suggestions right now though